src/core/server/status/plugins_status.ts

```diff
@@ -57,114 +97,139 @@ export class PluginsStatusService {
-          })
-          .map(([pName, status$]) =>
-            status$.pipe(map((status) => [pName, status] as [PluginName, ServiceStatus]))
-          );
-
-        return combineLatest(pluginStatuses).pipe(
-          map((statuses) => Object.fromEntries(statuses)),
-          distinctUntilChanged<Record<PluginName, ServiceStatus>>(isDeepStrictEqual)
-        );
-      })
+    return pluginData;
+  }
+
+  private getRootPlugins(): PluginName[] {
+    return Object.keys(this.pluginData).filter(
+      (plugin) => this.pluginData[plugin].dependencies.length === 0
+    );
+  }
+
+  private getOrderedPluginNames(): PluginName[] {
+    this.rootPlugins.forEach((plugin) => {
+      this.calculateDepthRecursive(plugin, 1);
+    });
+
+    return sortBy(Object.values(this.pluginData), ['depth', 'name']).map(({ name }) => name);
+  }
+
+  private calculateDepthRecursive(plugin: PluginName, depth: number) {
+    const pluginData = this.pluginData[plugin];
+    pluginData.depth = Math.max(pluginData.depth, depth);
+    const newDepth = depth + 1;
+    pluginData.reverseDependencies.forEach((revDep) =>
+      this.calculateDepthRecursive(revDep, newDepth)
     );
   }
 
-  /**
-   * Determines whether or not this plugin or any plugin in it's dependency tree have a custom status registered.
-   */
-  private anyCustomStatuses(plugin: PluginName): boolean {
-    if (this.pluginStatuses.get(plugin)) {
-      return true;
+  private updatePluginStatus(plugin: PluginName): void {
+    const newStatus = this.determinePluginStatus(plugin);
+    const pluginData = this.pluginData[plugin];
+    pluginData.derivedStatus = newStatus;
+
+    if (!pluginData.reportedStatus) {
+      // this plugin has NOT reported any status yet. Thus, its status is derived from its dependencies + core
+      this.pluginStatus[plugin] = newStatus;
+    }
+  }
+
+  private updatePluginsStatuses(plugins: PluginName[]): void {
+    const toCheck = new Set<PluginName>(plugins);
+    for (let i = 0; i < this.orderedPluginNames.length && toCheck.size > 0; ++i) {
+      const current = this.orderedPluginNames[i];
+      if (toCheck.has(current)) {
+        // update the current plugin status
+        this.updatePluginStatus(current);
+        // flag all its reverse dependencies to be checked
+        this.pluginData[current].reverseDependencies.forEach((revDep) => toCheck.add(revDep));
+      }
     }
+  }
+
+  private determinePluginStatus(plugin: PluginName): ServiceStatus {
+    const coreStatus: Array<[PluginName, ServiceStatus]> = Object.entries(this.coreStatus);
+    const depsStatus: Array<[PluginName, ServiceStatus]> = this.pluginData[plugin].dependencies.map(
+      (dependency) => [
+        dependency,
+        this.pluginData[dependency].reportedStatus || this.pluginData[dependency].derivedStatus,
+      ]
+    );
+
+    const newStatus = getSummaryStatus([...coreStatus, ...depsStatus], {
+      allAvailableSummary: `All dependencies are available`,
```

Comment:
nit: the same question about the necessity of the default value.

Reply:
See answer to the question above